packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -31,172 +25,51 @@ const MODAL_PLACE_ID = 'coz-harvest-modal-place'
+      .on(TWO_FA_MISMATCH_EVENT, this.displayTwoFAModal)
   }
 
   /**
-   * Ensure that a trigger will exist, with valid destination folder with
-   * permissions and references
-   * @return {Object} Trigger document
+   * Account submitting and trigger launching. Errors, events and statuses
+   * are handled by konnectorJob
+   * @param  {Object}  data [description]
    */
-  async ensureTrigger() {
-    const {
-      addPermission,
-      addReferencesTo,
-      createDirectoryByPath,
-      createTrigger,
-      statDirectoryByPath,
-      konnector,
-      t
-    } = this.props
+  async handleSubmit(data) {
+    const { konnector, konnectorJob, onLoginSuccess, onSuccess, t } = this.props
 
-    const { account, trigger } = this.state
-
-    if (trigger) {
-      return trigger
-    }
-
-    let folder
-
-    if (konnectors.needsFolder(konnector)) {
-      const path = `${t('default.baseDir')}/${konnector.name}/${slugify(
-        accounts.getLabel(account)
-      )}`
-
-      folder =
-        (await statDirectoryByPath(path)) || (await createDirectoryByPath(path))
-
-      await addPermission(konnector, konnectors.buildFolderPermission(folder))
-      await addReferencesTo(konnector, [folder])
-    }
-
-    return await createTrigger(
-      triggers.buildAttributes({
-        account,
-        cron: cron.fromKonnector(konnector),
-        folder,
-        konnector
-      })
-    )
-  }
-
-  /**
-   * Account save success handler
-   * @param  {Object}  account Created io.cozy.accounts document
-   * @return {Object}          io.cozy.jobs document, runned with account data
-   */
-  async handleAccountSaveSuccess(account) {
-    this.setState({ account })
-    const trigger = await this.ensureTrigger()
-    this.setState({ trigger })
-    this.accountWatcher = this.props.watchKonnectorAccount(account, {
-      onTwoFACodeAsked: this.handleTwoFACodeAsked,
-      onLoginSuccess: this.handleLoginSuccessState,
-      onLoginSuccessHandled: this.disableSuccessTimer
-    })
-    return await this.launch(trigger)
-  }
-
-  handleError(error) {
-    this.setState({
-      error,
-      status: ERRORED
-    })
-  }
-
-  handleTwoFACodeAsked(statusCode) {
-    const { status } = this.state
-    if (accounts.isTwoFANeeded(status)) return
-    // disable successTimeout since asked Two FA code
-    this.disableSuccessTimer()
-    this.setState({
-      status: statusCode
-    })
-  }
-
-  async handleSubmitTwoFACode(code) {
-    const { konnector, saveAccount } = this.props
-    const { account } = this.state
-    this.setState({
-      error: null,
-      status: RUNNING_TWOFA
-    })
-    try {
-      await saveAccount(konnector, accounts.updateTwoFaCode(account, code))
-      if (this.jobWatcher) this.jobWatcher.enableSuccessTimer(10000)
-    } catch (error) {
-      return this.handleError(error)
+    // prepare the connection, create account and trigger if needed
+    await konnectorJob.prepareConnection(konnector, data, t('default.baseDir'))
```

Comment:
Totally OK to externalizing business logic from TriggerManager, but it's not the KonnectorJob responsability to initialize every object. KonnectorJob should just laucnh a job for a given trigger, nothing more.

Trigger and account creation should be externalized in another business object.

Reply:
It's now moved to utility classes 👍